wp-includes/integrations/class-wp-service-worker-base-integration.php

```diff
@@ -18,18 +18,49 @@
 	 * @since 0.2
 	 * @var int
 	 */
-	protected $scope = WP_Service_Workers::SCOPE_FRONT;
+	protected $scope;
+
+	/**
+	 * Constructor.
+	 *
+	 * Sets the scope of the integration.
+	 *
+	 * @since 0.2
+	 */
+	public function __construct() {
+		$this->define_scope();
+	}
 
 	/**
 	 * Gets the scope this integration applies to.
 	 *
+	 * @since 0.2
+	 *
 	 * @return int Either WP_Service_Workers::SCOPE_FRONT, WP_Service_Workers::SCOPE_ADMIN, or
 	 *             WP_Service_Workers::SCOPE_ALL.
 	 */
 	public function get_scope() {
 		return $this->scope;
 	}
 
+	/**
+	 * Gets the priority this integration should be hooked into the service worker action with.
+	 *
+	 * @since 0.2
+	 *
+	 * @return int Hook priority. A higher number means a lower priority.
+	 */
+	public function get_priority() {
+		return 10;
```

Comment:
With the default priority being `10` for styles and scripts wouldn't it mean that if anyone would want to use the `wp_front_service_worker` or `wp_admin_service_worker` for opting in for precaching registered scripts they wouldn't be able to use the filters without knowing to use a priority lower that 10? Since that would be the way how to flag scripts and styles for precaching perhaps it would be better to do it so that it could be used without knowing to set a priority? Thoughts?

Reply:
@miina Good catch. At the moment it would indeed be necessary to use `wp_front_service_worker` / `wp_admin_service_worker` at a priority lower than 10 to flag scripts and styles for precaching because after that the `WP_Service_Worker_Scripts_Integration` / `WP_Service_Worker_Styles_Integration` logic will already have run.

On the other hand, I'm not sure though whether that is a big issue. I would think that eventually, scripts and styles should be flagged for being precached when they are registered/enqueued, which is `wp_enqueue_scripts` / `admin_enqueue_scripts`, and this should have executed before the `wp_front_service_worker` / `wp_admin_service_worker` hooks anyway.

Just to play safe, I think it would still make sense to have these two integrations run at a much higher priority value so that they are executed very late. This would mitigate the issue IMO.

Reply:
I changed the scripts and styles integrations to run at priority 10000 to address this.